modules/core/src/main/scala/playground/smithyutil/AddDynamicRefinements.scala

```diff
@@ -69,10 +76,12 @@ object AddDynamicRefinements extends (Schema ~> Schema) {
 
       case c: CollectionSchema[_, _] => collection(c)
       case m: MapSchema[_, _]        => m.reifyHint[api.Length]
-      // explicitly handling each remaining case, in order to get a "mising match" warning if the schema model changes
+      case e: EnumerationSchema[_] =>
+        enumSchema(e.asInstanceOf[EnumerationSchema[Int]])
```

Comment:
**issue:** I'm not sure why `enumSchema` needs to have an `Int` inside - and hence why we need casts here. I'll take a look..

Reply:
ok, looks like it wasn't necessary - removed